src/parcsr_ls/par_mgr_solve.c

```diff
@@ -252,43 +233,32 @@ hypre_MGRSolve( void               *mgr_vdata,
    /* check convergence within max_iter */
    if (iter == max_iter && tol > 0.)
    {
-      Solve_err_flag = 1;
       hypre_error(HYPRE_ERROR_CONV);
-   }
-
-   /*-----------------------------------------------------------------------
-    *    Print closing statistics
-    *    Add operator and grid complexity stats
-    *-----------------------------------------------------------------------*/
 
-   if (iter > 0 && init_resnorm)
-   {
-      conv_factor = hypre_pow((resnorm / init_resnorm), (fp_one / (HYPRE_Real) iter));
-   }
-   else
-   {
-      conv_factor = fp_one;
+      if (!my_id && (print_level & HYPRE_MGR_PRINT_INFO_SOLVE))
+      {
+         hypre_printf("\n\n==============================================");
+         hypre_printf("\n NOTE: Convergence tolerance was not achieved\n");
+         hypre_printf("      within the allowed %d iterations\n", max_iter);
+         hypre_printf("==============================================");
+      }
    }
 
-   if (print_level > 1)
+   if ((my_id == 0) && (tol > 0) && (print_level & HYPRE_MGR_PRINT_INFO_SOLVE))
```

Comment:
The logic seems to have changed here. This will only print if not using MGR as a preconditioner. Is that intended, and if so, why?

Reply:
Yes, logic has changed. New logic is: only print MGR's cycle data if the second bit of `print_level` is active (0x2). This is intended, because we have new print_level codes now (any number between 0 and 31) and we don't want to print MGR's cycle data always when `print_level > 1`. We want to print it only when the second bit is active, hence the bitwise AND with `HYPRE_MGR_PRINT_INFO_SOLVE`

Reply:
Sorry, Daniel! I read your msg too fast before... Now I see the point was the `tol > 0` check. I agree with you, it shouldn't be there. Fixed with [cef1f17](https://github.com/hypre-space/hypre/pull/976/commits/cef1f178dda6eba2b1378475d2e83d650f46eaf9)